src/components/action-popover/action-popover.component.js

```diff
@@ -0,0 +1,248 @@
+import React, {
+  useState, useCallback, useEffect, useRef
+} from 'react';
+import PropTypes from 'prop-types';
+import I18n from 'i18n-js';
+import {
+  Menu, MenuItemFactory, MenuButton, ButtonIcon, MenuItemIcon, MenuItemDivider
+} from './action-popover.style';
+import Events from '../../utils/helpers/events';
+import createGuid from '../../utils/helpers/guid';
+import OptionsHelper from '../../utils/helpers/options-helper';
+
+const ActionPopover = ({ children, id }) => {
+  const [isOpen, setOpen] = useState(false);
+  const [focusIndex, setFocusIndex] = useState(0);
+  const [childrenWithRef, setChildrenWithRef] = useState();
+  const [items, setItems] = useState();
+  const [guid] = useState(createGuid());
+  const button = useRef();
+
+
```

Comment:
Sorry, one last thing - an extra / unnecessary new line has crept in here! Once that's gone I'll approve 👍